pkg/controller/compliance/compliance_controller.go

```diff
@@ -184,11 +184,17 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 		return reconcile.Result{}, err
 	}
 
-	if err = utils.CheckLicenseKey(ctx, r.client); err != nil {
+	license, err := utils.CheckLicenseKey(ctx, r.client)
+	if err != nil {
 		r.status.SetDegraded("License not found", err.Error())
 		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
 	}
 
+	if !utils.IsComplianceFeatureActive(license) {
```

Comment:
If compliance feature is not active, my guess would be that:
- we probably want to throw an error to make people remove compliance
- we set r.status to degraded